diff --git a/ctn91xx_pci.c b/ctn91xx_pci.c
index fde14ce..3540f85 100644
--- a/ctn91xx_pci.c
+++ b/ctn91xx_pci.c
@@ -44,7 +44,7 @@ static int ctn91xx_register(struct pci_dev *pdev, const struct pci_device_id *en
     }

     /* Try to init DMA */
-    if(pci_set_dma_mask(pdev, DMA_32BIT_MASK)) {
+    if(pci_set_dma_mask(pdev, DMA_BIT_MASK(32))) {
         ERROR("No suitable DMA mask available.");
     }
